src/script_lua.c

```diff
@@ -1269,6 +1269,37 @@ sds luaGetStringSds(lua_State *lua, int index) {
     return str_sds;
 }
 
+char *luaGetStringCStr(lua_State *lua, int index) {
+    if (!lua_isstring(lua, index)) {
+        return NULL;
+    }
+
+    size_t len;
+    const char *str = lua_tolstring(lua, index, &len);
+
+    size_t n = 0;
+    for (size_t i = 0; i < len; i++) {
+        if (str[i] == '\0') {
+            n++;
+        }
+    }
+
+    char *ret_str = zcalloc(len + n + 1);
+
+    n = 0;
+    for (size_t i = 0; i < len; i++) {
+        if (str[i] != '\0') {
+            ret_str[n] = str[i];
+        } else {
+            ret_str[n++] = '\\';
+            ret_str[n] = '0';
+        }
+        n++;
+    }
+
+    return ret_str;
+}
```

Comment:
The string values stored in the Lua stack may contain null characters, and there is a test case that is testing an invalid shebang line in a lua script by adding null characters in the middle of the shebang line. If we return the string just until the first null, the validation will fail.

To solve this problem I decided to escape the null characters in this function. But now that I'm looking again at it, what I should have done was, instead of perform the escaping of null chars, return the string pointer along with the length of the string, and let the callers deal with the null chars problem.

I'm going to change this code to return the string length as well.